cloudconfig.go

```diff
@@ -0,0 +1,71 @@
+package cloudconfig
+
+import (
+	"bytes"
+	"compress/gzip"
+	"encoding/base64"
+	"html/template"
+	"io/ioutil"
+
+	"github.com/giantswarm/awstpr"
+	tpraws "github.com/giantswarm/awstpr/aws"
+	"github.com/giantswarm/clustertpr/node"
+)
+
+type AWSNode struct {
```

Comment:
I don't understand the problem. The interface is the template. The template contains a cloud config. The parts that are the same everywhere are hard coded in the template. The parts that need to be dynamically injected will be injected via template data. This is the interface. Lets talk if something is not clear.